mne/beamformer/_compute_beamformer.py

```diff
@@ -253,92 +199,108 @@ def _compute_beamformer(G, Cm, reg, n_orient, weight_norm, pick_ori,
                 'Singular matrix detected when estimating spatial filters. '
                 'Consider reducing the rank of the forward operator by using '
                 'reduce_rank=True.')
+        del Gk_s
 
     # rank reduction of the lead field
     if reduce_rank:
         Gk = _reduce_leadfield_rank(Gk)
 
-    # Compute numerator of beamformer formula, G.T @ Cm_inv
+    #
+    # 1. Compute filter
+    #
+
+    # Compute numerator of beamformer formula:
+    # Wk = G.T @ Cm_inv
     Wk = np.matmul(Gk.transpose(0, 2, 1), Cm_inv[np.newaxis])
 
-    if (inversion == 'matrix' and pick_ori == 'max-power' and
-            weight_norm in ['unit-noise-gain', 'nai']):
-        # In this case, take a shortcut to compute the filter
-        _normalized_weights(Wk, Gk, Cm_inv_sq, reduce_rank, nn, sk)
-    else:
-        # Compute power at the source
-        Ck = np.matmul(Wk, Gk)  # np.dot for each source
-
-        # Normalize the spatial filters
-        if n_orient > 1:
-            # Free source orientation
-            if inversion == 'single':
-                # Invert for each dipole separately using plain division
-                diags = np.diagonal(Ck, axis1=1, axis2=2)
-                assert not reduce_rank   # guaranteed above
-                with np.errstate(divide='ignore'):
-                    diags = 1. / diags
-                # set the diagonal of each 3x3
-                norm = np.zeros((n_sources, n_orient, n_orient), Ck.dtype)
-                for k in range(n_sources):
-                    norm[k].flat[::4] = diags[k]
-            elif inversion == 'matrix':
-                assert Ck.shape[1:] == (3, 3)
-                # Invert for all dipoles simultaneously using matrix
-                # inversion.
-                norm = _pos_semidef_inv(Ck, reduce_rank)
-            # Reapply source covariance after inversion
-            norm *= sk[:, :, np.newaxis]
-            norm *= sk[:, np.newaxis, :]
-        else:  # n_orient == 1
-            assert Ck.shape[1:] == (1, 1)
-            # Fixed source orientation
+    #
+    # 2. Compute max power orientation
+    #
+    if pick_ori == 'max-power':
+        # Compute the power
+        if inversion == 'matrix':
+            # Compute power by applying the spatial filters to the cov matrix
+            mult = Wk
+        else:
+            assert inversion == 'single'
+            # First make the filters unit gain, then apply them to the
+            # cov matrix to compute power.
+            mult = Wk / np.linalg.norm(Wk, axis=2, keepdims=True)
```

Comment:
I don't think this works here, the code snippet comes from further down, where `Wk` already refers to a later step in filter calculation. I will see if I can fix it!